src/sentry/conf/server.py

```diff
@@ -31,18 +31,32 @@
 def env(key, default='', type=None):
     "Extract an environment variable for use in configuration"
 
-    if 'SENTRY_RUNNING_UWSGI' in os.environ:
-        # We do this so when the process forks off into uwsgi
-        # we want to actually be popping off values. This is so that
-        # at runtime, the variables aren't actually available.
-        fn = os.environ.pop
-    else:
-        fn = os.environ.get
+    # First check an internal cache, so we can `pop` multiple times
+    # without actually losing the value.
+    try:
+        rv = env._cache[key]
```

Comment:
The reason that design decision was done in the first place is it's common practice to not let `os.environ` leak to potentially sensitive information to subprocesses. And it's pretty common to pass along `os.environ.copy()` when creating subprocesses. If we were to have a secret value here, I'm trying to explicitly pop it when reading it.

So it's not so much that it's in memory, it's just in memory in a place that's less likely to be leaked to things that would inadvertantly get it.